source/Core/ModuleList.php

```diff
@@ -415,93 +415,22 @@ protected function _removeFromDisabledModulesArray($aDeletedExtIds)
     }
 
     /**
-     * Removes extension from modules paths array
+     * Removes extension from given modules array
      *
-     * @param array $aDeletedModule deleted extensions ID's
+     * @param string $key module array key
+     * @param array  $aDeletedModule deleted extensions ID's
      */
-    protected function _removeFromModulesPathsArray($aDeletedModule)
+    protected function _removeFromModulesArray($key, $aDeletedModule)
     {
-        $aModulePaths = $this->getModulePaths();
-
-        foreach ($aDeletedModule as $sDeletedModuleId) {
-            if (isset($aModulePaths[$sDeletedModuleId])) {
-                unset($aModulePaths[$sDeletedModuleId]);
-            }
-        }
-
-        $this->getConfig()->saveShopConfVar('aarr', 'aModulePaths', $aModulePaths);
-    }
-
-    /**
-     * Removes extension from modules versions array
-     *
-     * @param array $aDeletedModule deleted extensions ID's
-     */
-    protected function _removeFromModulesVersionsArray($aDeletedModule)
-    {
-        $aModuleVersions = $this->getModuleVersions();
-
-        foreach ($aDeletedModule as $sDeletedModuleId) {
-            if (isset($aModuleVersions[$sDeletedModuleId])) {
-                unset($aModuleVersions[$sDeletedModuleId]);
-            }
-        }
-
-        $this->getConfig()->saveShopConfVar('aarr', 'aModuleVersions', $aModuleVersions);
-    }
-
-    /**
-     * Removes extension from modules events array
-     *
-     * @param array $aDeletedModule deleted extensions ID's
-     */
-    protected function _removeFromModulesEventsArray($aDeletedModule)
-    {
-        $aModuleEvents = $this->getModuleEvents();
+        $array = $this->{'getModule' . $key}();
```

Comment:
Could we introduce some "abstract" method here? Something like "getModuleConfigParametersByKey". 
I do not like the fact, that it is not clear, which methods will be called here and if this methods is defined at all. Code analysis tools like CodeSniffer and MessDetector cannot help avoiding calls to undefined methods with dynamic method names.